src/map.cpp

```diff
@@ -1582,7 +1582,7 @@ MapBlock *ServerMap::getBlockOrEmerge(v3s16 p3d)
 {
 	MapBlock *block = getBlockNoCreateNoEx(p3d);
 	if (block == NULL)
-		m_emerge->enqueueBlockEmerge(PEER_ID_INEXISTENT, p3d, false);
+		m_emerge->enqueueBlockEmerge(PEER_ID_INEXISTENT, p3d, true);
```

Comment:
in this case it might be that `getBlockOrEmerge` is only used in the place we care about and changing this is fine, but I think the "will this method emerge with generate or not" situation is too opaque here.
suggestion: add `bool generate` to method

Reply:
Not sure I follow. You're saying there might be a use in the future where we want to get blocks in the map but not allow generating?

There are three places where EmergeManager::enqueueBlockEmerge within all of MT:
1. `RemoteClient::GetNextBlocks` here we do not want to generate when we're past the generate distance
2. `Server::handleCommand_Interact` where apparently we would load a block from disk but not generate it.
3. From here.

In general I feel we should not distinguish between loading and generating anywhere except in case (1) above, for a player it is immaterial whether a blocks is being generate or "cached" in the database.

Not opposed to add a flag at all. Just wondering what the motivation is.

Reply:
> You're saying there might be a use in the future where we want to get blocks in the map but not allow generating?

Maybe.
The primary motivation is to avoid surprises in the code.